### PR TITLE
Prefer StringBuilder.Append(char) over Append(string)

### PR DESCRIPTION
**Pull request by stephentoub**
The char overload has a much simpler/faster implementation.  And if nothing else this increases our consistency across the codebase, as we already use the char overload in a bunch of places.

### CODE REVIEW

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs [hunk 1]
```diff
@@ -420,7 +420,7 @@ private string GetMessageLibraryNames(string libRegKey)
                         result.Append(owner.MachineName);
                         result.Append(@"\");
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs]: Looks like one got missed?

Comment (AaronRobinsonMSFT) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs]: ~~`result.Append(@"\");`~~ -> `result.Append('\\');`

File: src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs [hunk 2]
```diff
@@ -66,7 +66,7 @@ public void Write(StringBuilder strBuilder, DocPosition docPos, AncestralNamespa
             {
                 anc.GetNamespacesToRender(this, attrListToRender, nsListToRender, nsLocallyDeclared);
 
-                strBuilder.Append("<" + Name);
+                strBuilder.Append('<').Append(Name);
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs]: Just to confirm - this won't bug out if _Name_ suddenly becomes null, right?

**Review by GrabYourPitchforks — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs (hunk 2 above)
Comment (stephentoub) [reply src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs]: > this won't bug out if Name suddenly becomes null, right?

Correct.
https://github.com/dotnet/runtime/blob/2d6b77fa4eb4811d33f23d0183a12cf2b5d4ab6e/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L807

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs]: Yup.  Incomplete regex :)  Hopefully when actually implemented in https://github.com/dotnet/runtime/issues/30740, that rule will do better. (One of my motivations here was to see how common this actually was, to help understand the benefit of such a rule.)

Comment (stephentoub) [reply src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs]: Fixed. Thanks.

**Review by GrabYourPitchforks — APPROVED** [review]

**Pull request merged**